src/GraphQL/Types/Schema.cs

```diff
@@ -265,7 +259,7 @@ private GraphTypesLookup CreateTypesLookup()
             return GraphTypesLookup.Create(
                 types,
                 _directives,
-                type => DependencyResolver.Resolve(type) as IGraphType,
+                type => Services.GetService(type) as IGraphType,
```

Comment:
Consider using `GetRequiredService` extension method where necessary instead of `GetService`. `GetService` [**can**](https://docs.microsoft.com/en-us/dotnet/api/system.iserviceprovider.getservice?view=netcore-2.1#System_IServiceProvider_GetService_System_Type_) return null.

```c#
    public static object GetRequiredService(this IServiceProvider provider, Type serviceType)
    {
      if (provider == null)
        throw new ArgumentNullException(nameof (provider));
      if (serviceType == (Type) null)
        throw new ArgumentNullException(nameof (serviceType));
      object service = provider.GetService(serviceType);
      if (service != null)
        return service;
      throw new InvalidOperationException($"Required service for type {serviceType} not found"));
    }
```

This will make it possible to find an error at an early stage, before a possible `null` transmission to the underlying code.

Reply:
Agreed - I didn't want to include that extra dependency but I can easily add that as an internal extension method.

Reply:
Yes, no extra dependency, just a little helper/extension method.